### PR TITLE
Further refactoring of unit code

### PR DESCRIPTION
**Pull request by royfalk**
- Split master_part_list to multiple files
- Make ship into a separate super class of unit
- Move additional methods to movable

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- Ship was damaged after respawning

**Note: this required the complementing changes to data files.**

### CODE REVIEW

File: engine/src/cmd/energetic.h
Comment (BenjamenMeyer) [inline engine/src/cmd/energetic.h:103]: everything prior to this is public, so if we're going to change it then we should just drop it.

File: engine/src/cmd/movable.cpp
Comment (BenjamenMeyer) [reply engine/src/cmd/movable.cpp:479]: Please fix this warning so we don't lose data or improperly cut-off data and lose precision.

File: engine/src/cmd/movable.cpp
Comment (BenjamenMeyer) [inline engine/src/cmd/movable.cpp:697]: Perhaps we need to change compiler flags instead of opting out platforms from this logic check?
https://developer.apple.com/forums/thread/656056

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]

File: engine/src/cmd/movable.cpp
Comment (royfalk) [reply engine/src/cmd/movable.cpp:697]: I dug a bit into this and I don't know why it's here. The original [commit](https://github.com/vegastrike/Vega-Strike-Engine-Source/commit/58be34265c8382c848eadfce58aea7d35cd391e0) is from 2006. OSX was introduced in 2001. Apple switched to Intel computers in 2006 but that should have made it more consistent with Linux, not break it. There's a good chance that if we get the game running on Mac, we won't need this.

Comment (BenjamenMeyer) [reply engine/src/cmd/movable.cpp:697]: @royfalk if we can get rid of it that'd be awesome!

Comment (stephengtuggy) [reply engine/src/cmd/movable.cpp:697]: @BenjamenMeyer I'd like to recommend that we leave this code as is for now. Then once we get #773 reviewed and merged (hint, hint) then we can come back and revisit this topic.

Comment (BenjamenMeyer) [reply engine/src/cmd/movable.cpp:697]: @stephengtuggy that's fine. It'd be nice to simplify this so it's the same on all platforms or easily swappable via compiler flags instead.  However we get there - or whichever PR gets us there - doesn't matter to me. Thanks!

File: engine/src/cmd/movable.cpp
Comment (stephengtuggy) [reply engine/src/cmd/movable.cpp:479]: I agree that it's probably best to fix this warning.

File: engine/src/cmd/movable.cpp
Comment (royfalk) [reply engine/src/cmd/movable.cpp:697]: Moved to #787. Resolving conversation.

**Comment by royfalk** [comment]
I think I did everything needed here. Can I get an approval?

**Comment by stephengtuggy** [comment]
@royfalk You do realize that this includes the changes from the big #754 refactor, right? I don't think those are ready to go into the `master` branch yet, unless I am mistaken.

**Comment by stephengtuggy** [comment]
I would be okay with merging these changes into a 754-specific working branch.

**Review by stephengtuggy — APPROVED** [review]
OK, my mistake. Upon checking the commit list more carefully, I realized that this PR includes the commit(s) rolling back 754, as well.

Looks good. Approving and merging.

**Comment by royfalk** [comment]
> OK, my mistake. Upon checking the commit list more carefully, I realized that this PR includes the commit(s) rolling back 754, as well.
> 
> Looks good. Approving and merging.

Note the warning at the top. It is coupled to a change in assets that I still need to submit. Once approved, I'll push it.

**Comment by stephengtuggy** [comment]
Actually, I can't merge this until @BenjamenMeyer re-reviews and approves it. Crap. Ben, would you be so kind as to review this again?

**Comment by BenjamenMeyer** [comment]
Been away this weekend will take a look as soon as I can

**Review by BenjamenMeyer — APPROVED** [review]

**Comment by royfalk** [comment]
I'm not sure why git keeps asking me to rebase the same changes over and over. 
I finally forced push these in the hopes this will do the trick.

**Pull request merged**
